agent/engine/docker_task_engine.go

```diff
@@ -1417,6 +1417,27 @@ func (engine *DockerTaskEngine) provisionContainerResources(task *apitask.Task,
 	}
 }
 
+// checkTearDownPauseContainer idempotently tears down the pause container network when the pause container's known
+//or desired status is stopped.
+func (engine *DockerTaskEngine) checkTearDownPauseContainer(task *apitask.Task) {
+	for _, container := range task.Containers {
```

Comment:
instead of iterating through all containers, can we check if task is awsvpc enabled (GetENI != nil) and return quickly?

Reply:
is it time or readability that would be improved? In terms of time, there can be a max of 10 containers so we would gain a few nanoseconds :p

Reply:
as a practice, just like to return quickly wherever possible :) also, containers count could be increased in the future

Reply:
Fair enough. changed.